app/src/main/java/ch/epfl/sdp/blindly/main_screen/match/my_matches/MyMatchesAdapter.kt

```diff
@@ -0,0 +1,148 @@
+        val matchedName: TextView = view.findViewById(R.id.matchedUserName)
+        val userNameLayout: LinearLayout = view.findViewById(R.id.userNameLayout)
+        val expandableChatAndMapLayout: RelativeLayout = view.findViewById(R.id.chatAndMapLayout)
+        val chatButton: AppCompatImageButton = view.findViewById(R.id.chatButton)
+        val mapButton: AppCompatImageButton = view.findViewById(R.id.mapButton)
+
+        init {
+            userNameLayout.setOnClickListener(this)
+            expandableChatAndMapLayout.visibility = View.GONE
+        }
+
+        override fun onClick(v: View?) {
+            if (adapterPosition != RecyclerView.NO_POSITION) {
+                listener.onItemClick(adapterPosition)
+            }
+        }
+    }
+
+    /**
+     * Collapses all record layouts in the view.
+     */
+    private fun collapseLayouts() {
+        for (i in 0 until my_matches.size) {
+            if (my_matches[i].isExpanded) {
+                val viewHolder = viewHolderList[i]
+                RecordAnimations.collapse(viewHolder.expandableChatAndMapLayout)
+                my_matches[i].isExpanded = false
+            }
+        }
+    }
+
+    /**
+     * Creates new views for each item of the list.
+     * This function is invoked by the layout manager.
+     *
+     * @param viewGroup the group where the ViewHolder belongs
+     * @param viewType the type of view
+     * @return a new view, which defines the UI of the list items
+     */
+    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
+        val view =
+            LayoutInflater.from(viewGroup.context).inflate(R.layout.my_matched_user_item, null)
+        viewHolderList.add(ViewHolder(view))
+        return ViewHolder(view)
+    }
+
+    /**
+     * Binds all the content to the view, and handles click listeners.
+     * This function is invoked by the layout manager.
+     *
+     * @param viewHolder the ViewHolder of the current item
+     * @param position the position of the item in the RecyclerView
+     */
+    @RequiresApi(Build.VERSION_CODES.N)
+    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
+        viewHolder.matchedName.text = my_matches[position].name
+
+        /*
+         * When the layout containing the name is clicked, the layout expands or
+         * collapses, and the layout is reset so that it is brand new when re-opened.
+         */
+        viewHolder.userNameLayout.setOnClickListener {
+            val notIsExpanded = !my_matches[position].isExpanded
+            toggleLayout(notIsExpanded, viewHolder.expandableChatAndMapLayout)
+            my_matches[position].isExpanded = notIsExpanded
+        }
+
+
+        viewHolder.chatButton.setOnClickListener {
+            val intent = Intent(context, ChatActivity::class.java)
+            val bundle = bundleOf("matchedId" to my_matches[position].uid)
```

Comment:
Maybe create a const for the "matchedId"

Reply:
Yep, will do.